app/src/main/java/com/mapbox/maps/testapp/examples/markersandcallouts/ViewAnnotationAnimationActivity.kt

```diff
@@ -0,0 +1,1783 @@
+package com.mapbox.maps.testapp.examples.markersandcallouts
+
+import android.animation.Animator
+import android.animation.AnimatorListenerAdapter
+import android.animation.TypeEvaluator
+import android.animation.ValueAnimator
+import android.animation.ValueAnimator.AnimatorUpdateListener
+import android.graphics.Color
+import android.os.Bundle
+import android.view.ViewGroup.LayoutParams.WRAP_CONTENT
+import android.widget.Button
+import android.widget.FrameLayout
+import androidx.appcompat.app.AppCompatActivity
+import com.mapbox.geojson.FeatureCollection
+import com.mapbox.geojson.LineString
+import com.mapbox.geojson.Point
+import com.mapbox.maps.CameraOptions
+import com.mapbox.maps.Style
+import com.mapbox.maps.ViewAnnotationOptions
+import com.mapbox.maps.extension.style.layers.generated.lineLayer
+import com.mapbox.maps.extension.style.sources.generated.geoJsonSource
+import com.mapbox.maps.extension.style.style
+import com.mapbox.maps.testapp.databinding.ActivityViewAnnotationAnimationBinding
+import com.mapbox.maps.viewannotation.ViewAnnotationManager
+import com.mapbox.turf.TurfConstants.UNIT_DEFAULT
+import com.mapbox.turf.TurfMeasurement
+import kotlin.math.roundToLong
+
+/**
+ * Example how to animate a view annotations on top of a route.
+ */
+class ViewAnnotationAnimationActivity : AppCompatActivity() {
+
+  private lateinit var viewAnnotationManager: ViewAnnotationManager
+  private var routeCoordinateList = mutableListOf<Point>()
+  private lateinit var button: Button
+  private var totalLength = 0.0
+  private var count = 0
+  private var currentAnimator: Animator? = null
+
+  override fun onCreate(savedInstanceState: Bundle?) {
+    super.onCreate(savedInstanceState)
+    val binding = ActivityViewAnnotationAnimationBinding.inflate(layoutInflater)
+    setContentView(binding.root)
+
+    // calculate the route length, get coordinates from route geometry
+    val featureCollection = FeatureCollection.fromJson(ROUTE_GEOJSON)
+    val lineString = featureCollection.features()!![0].geometry() as LineString
+    routeCoordinateList = lineString.coordinates()
+    totalLength = TurfMeasurement.length(lineString, UNIT_DEFAULT)
+
+    // initialize the mapview
+    binding.mapView.getMapboxMap().apply {
+      loadStyle(
+        style(Style.LIGHT) {
+          // source for displaying the route
+          +geoJsonSource(SOURCE_ID) {
+            featureCollection(featureCollection)
+          }
+          // layer for displaying the route
+          +lineLayer(LINE_ID, SOURCE_ID) {
+            lineColor(Color.parseColor("#F13C6E"))
+            lineWidth(4.0)
+          }
+        }
+      ) {
+        // center camera around SF airport
+        setCamera(
+          CameraOptions.Builder()
+            .center(Point.fromLngLat(-122.3915, 37.6177))
+            .zoom(11.0)
+            .build()
+        )
+
+        // create a dummy view to animate
+        button = Button(binding.mapView.context)
```

Comment:
could re-use `item_callout_view.xml` from other examples and even e.g. update text dynamically when moving with new lat / lng.